### PR TITLE
chore: Bump rustls from 0.23.15 to 0.23.19 to fix RUSTSEC-2024-0399

### PR DESCRIPTION
**Pull request by sbernauer**
# Description

Fixes https://github.com/stackabletech/operator-rs/issues/916
Part of https://github.com/stackabletech/issues/issues/677

Note: I ran `cargo update` to get the changes.

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes

```[tasklist]
# Author
- [ ] Changes are OpenShift compatible
- [ ] CRD changes approved
- [ ] Integration tests passed (for non trivial changes)
```

```[tasklist]
# Reviewer
- [ ] Code contains useful comments
- [ ] (Integration-)Test cases added
- [ ] Documentation added or updated
- [ ] Changelog updated
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
```

### CODE REVIEW

**Comment by adwk67** [comment]
I'm getting some test failures on compile:
`test tests/default/fail/skip_from_all.rs ... mismatch`

**Comment by sbernauer** [comment]
Can you please try `rustup component add rust-src` and re-run?
CC @Techassi @NickLarsenNZ

**Comment by sbernauer** [comment]
(It's unrelated to this PR, main should fail for you as well)

**Comment by adwk67** [comment]
> rustup component add rust-src

No change. I'm running the tests and get

```
ACTUAL OUTPUT:
┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈
error[E0308]: mismatched types
  --> tests/default/fail/skip_from_version.rs:23:32
   |
23 |     let foo_v1 = v1::Foo::from(foo_v1beta1);
   |                  ------------- ^^^^^^^^^^^ expected `main::v1::Foo`, found `v1beta1::Foo`
   |                  |
   |                  arguments to this function are incorrect
   |
   = note: `v1beta1::Foo` and `main::v1::Foo` have similar names, but are actually distinct types
note: `v1beta1::Foo` is defined in module `crate::main::v1beta1` of the current crate
  --> tests/default/fail/skip_from_version.rs:4:5
   |
4  | /     #[versioned(
5  | |         version(name = "v1alpha1"),
6  | |         version(name = "v1beta1", skip(from)),
7  | |         version(name = "v1")
8  | |     )]
   | |______^
note: `main::v1::Foo` is defined in module `crate::main::v1` of the current crate
  --> tests/default/fail/skip_from_version.rs:4:5
   |
4  | /     #[versioned(
5  | |         version(name = "v1alpha1"),
6  | |         version(name = "v1beta1", skip(from)),
7  | |         version(name = "v1")
8  | |     )]
   | |______^
note: associated function defined here
  --> $RUST/core/src/convert/mod.rs
   |
   |     fn from(value: T) -> Self;
   |        ^^^^
   = note: this error originates in the attribute macro `versioned` (in Nightly builds, run with -Z macro-backtrace for more info)
```

**Comment by Techassi** [comment]
Yes @adwk67 errors are unrelated to the change. Make sure you have the `rust-src` component installed and use the latest Rust toolchain (specified in `rust-toolchain.toml`). Also make sure to run `cargo clean` to remove any cached files.

**Review by Techassi — CHANGES_REQUESTED** [review]
Please add an entry in the changelog. Also mention _how_ you bumped the dependency in the PR description, so `cargo update -p rustls` for example.

**Comment by adwk67** [comment]
> Yes @adwk67 errors are unrelated to the change. Make sure you have the `rust-src` component installed and use the latest Rust toolchain (specified in `rust-toolchain.toml`). Also make sure to run `cargo clean` to remove any cached files.

I actually had to remove the toolchain component with:
`rustup component remove rust-src --toolchain 1.82.0`
and then all tests were green.

**Comment by Techassi** [comment]
> I actually had to remove the toolchain component

Mhm... That's curious. Not quite sure why that's the case. I can investigate this a little more if we see the need for it.

File: crates/stackable-operator/CHANGELOG.md [hunk 1]
```diff
@@ -7,8 +7,10 @@ All notable changes to this project will be documented in this file.
 ### Changed
 
 - Replace unmaintained `derivative` crate with `educe` ([#907]).
+- Bump dependencies, notably rustls 0.23.15 to 0.23.19 to fix RUSTSEC-2024-0399 ([#917]).
```
Comment (Techassi) [inline crates/stackable-operator/CHANGELOG.md:10]: suggestion: Link to the advisory.

```suggestion
- Bump dependencies, notably rustls 0.23.15 to 0.23.19 to fix [RUSTSEC-2024-0399] ([#917]).
```

File: crates/stackable-operator/CHANGELOG.md [hunk 2]
```diff
@@ -7,8 +7,10 @@ All notable changes to this project will be documented in this file.
...
 
 [#907]: https://github.com/stackabletech/operator-rs/pull/907
+[#917]: https://github.com/stackabletech/operator-rs/pull/917
```
Comment (Techassi) [inline crates/stackable-operator/CHANGELOG.md:13]:
```suggestion
[#917]: https://github.com/stackabletech/operator-rs/pull/917
[RUSTSEC-2024-0399]: https://rustsec.org/advisories/RUSTSEC-2024-0399
```

**Review by Techassi — APPROVED** [review]

**Pull request merged**